SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -580,6 +580,7 @@ void USpatialNetDriver::CreateAndInitializeCoreClassesAfterStartup()
 		}
 
 		ClientConnectionManager = MakeUnique<SpatialGDK::ClientConnectionManager>(SystemEntitySubview, this);
+		PlayerSpawner->StartProcessingRequests();
```

Comment:
How long after creation does this generally happen and are we expecting that clients will be timing out before this point? It looks like we do retry sending the spawn requests so is this going to result in multiple spawns?

Reply:
Putting things in a map or set might help but then it's rolling the dice on when it times out. We could also deliberately fail commands that arrive too early.

Reply:
Honestly I think this is only actually a problem when running in editor because the clients are started at the same time as the servers, whereas any other flow would most likely having a server registration step that would prevent clients connecting too early. We also have this (https://github.com/spatialos/UnrealGDK/blob/38337d06e22c5e0fd49e08eec5a71752c411a8cf/SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialPlayerSpawner.h#L88) which prevent double spawns from happening.

Reply:
That only works if the spawn wasn't forwarded doesn't it?

Reply:
Actually I'm slightly confused by this being in the create core classes bit. Shouldn't be be processing stuff as part of the startup flow?

Reply:
>That only works if the spawn wasn't forwarded doesn't it?

No it's maintained on both the server auth over the spawner, and the server that processed the actual spawn.

>Actually I'm slightly confused by this being in the create core classes bit. Shouldn't be be processing stuff as part of the startup flow?

It's there because all that it requires is the map to be loaded and the `ClientConnectionManager` spawned. It could potentially be moved as part of Dmitrii's startup ordering, but I don't think worth doing here.

Reply:
We don't want to make this sync up with the point at which we write to state that clients can connect then? Isn't that going to get  confusing?

Reply:
I've moved the call to Dmitrii's new startup flow